### PR TITLE
LOG-2123: Refactor some syslog e2e tests to functional framework

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* Moves syslog functional test functionality into the functional framework from e2e framework
* Leaves syslog e2e tests in place that verify secure versus non-secure forward to syslog
* Generally removes tests that were questionably written

cc @alanconway @vimalk78 

### Links
* https://issues.redhat.com/browse/LOG-2123

### CODE REVIEW

**Review by alanconway — APPROVED** [review]
/lgtm

File: test/framework/functional/fluentd/deploy.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func (c *FluentdCollector) BuildCollectorContainer(b *runtime.ContainerBuilder,
 
 func (c *FluentdCollector) IsStarted(logs string) bool {
 	// if fluentd started successfully return success
-	return strings.Contains(logs, "flush_thread actually running")
+	return strings.Contains(logs, "fluentd worker is now running worker")
```
Comment (syedriko) [inline test/framework/functional/fluentd/deploy.go:53]: too many workers

**Pull request merged**